backend/apps/notificationpref/factories.py

```diff
@@ -0,0 +1,17 @@
+import factory
+from apps.notificationpref.models import UserNotificationPreference
+from apps.users.factories import BaseUserFactory
+
+class UserNotificationPreferenceFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = UserNotificationPreference
+
+    user = factory.SubFactory(BaseUserFactory)
+
+    id = factory.Sequence(lambda n: 1000 + n)
```

Comment:
If you remove id then it should be auto generated by the AutoField in the model itself